### PR TITLE
Configure e2e tests in CI 

### PR DESCRIPTION
**Pull request by nvisal1**
This PR adds Circle CI configuration for e2e tests. 

Required by #759 
Required by #643 
Required by #642

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -12,25 +12,42 @@ jobs:
...
             - ./
   test:
     <<: *defaults
     steps:  
       - attach_workspace : 
           at: ~/clark-client
+      - run:
+          name: Clone e2eBox Repository 
+          command: >-
+           ls && cd .. && git clone https://${github_username}:${github_password}@github.com/nvisal1/CLARK-e2eBox.git
```
Comment (gshaw1997) [inline .circleci/config.yml]: Is this ls just for testing?

File: .circleci/config.yml [hunk 2]
```diff
@@ -12,25 +12,42 @@ jobs:
...
+      - run:
+          name: Install Docker client
```
Comment (gshaw1997) [inline .circleci/config.yml]: Use `setup_remote_docker` instead.

https://circleci.com/docs/2.0/building-docker-images/

File: .circleci/config.yml [hunk 3]
```diff
@@ -12,25 +12,42 @@ jobs:
...
+          command: |
+            set -x
+            VER="17.03.0-ce"
+            curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/stable/x86_64/docker-$VER.tgz
+            tar -xz -C /tmp -f /tmp/docker-$VER.tgz
+            mv /tmp/docker/* /usr/bin
+      - run: 
+          name: Install Docker Compose
```
Comment (gshaw1997) [inline .circleci/config.yml]: This will need to happen before setting up remote docker engine

https://circleci.com/docs/2.0/docker-compose/

File: cypress/integration/browse.spec.js [hunk 4]
```diff
@@ -13,8 +13,9 @@ describe('Browse', () => {
     beforeEach(() => {
         // Return to home page before each test
         cy.fixture('route.json').then((route) => {
+            console.log(route)
             home = route;
```
Comment (gshaw1997) [inline cypress/integration/browse.spec.js]: `home` should probably be renamed to be more indicative of the values it contains, seems to store more than just home info

File: src/environments/environment.ts [hunk 5]
```diff
@@ -5,10 +5,10 @@
 
 export const environment = {
   production: false,
-  apiURL: 'http://localhost:3000',
+  apiURL: 'http://clark-gateway:3000',
```
Comment (gshaw1997) [inline src/environments/environment.ts]: Are these temp values? How would this affect local dev?

Comment (DonSeannelly) [reply src/environments/environment.ts]: This is going to force it to be run in a docker network. Can we get around this through any build commands with the Angular CLI?

If not, maybe we need to add some proxy setup to the image.

File: cypress/integration/browse.spec.js [hunk 6]
```diff
@@ -13,8 +13,9 @@ describe('Browse', () => {
     beforeEach(() => {
         // Return to home page before each test
         cy.fixture('route.json').then((route) => {
+            console.log(route)
```
Comment (nrwinner) [inline cypress/integration/browse.spec.js]: Why was this added?
